docs/usage/integration.md

```diff
@@ -0,0 +1,413 @@
+  - Operations within a request are executed in sequential order.
+  - Operations within a request do not interleave with other requests.
+    (TODO: Can timeouts interleave batches, or should we shift the batch so that timeouts land
+    entirely before/after?)
+- **Operations**
+  - Once committed, an operation will always be committed — the cluster's state never backtracks.
+  - [Accounts](../reference/accounts.md) are immutable — once created, they are never modified
+    (excluding balance fields, which are modified by transfers).
+  - [Transfers](../reference/transfers.md) are immutable — once created, they are never modified.
+  - Transfer [timeouts](../reference/transfers.md#timeout) are deterministic, driven by the
+    cluster's timestamp.
+  - Object timestamps are cluster-unique. For objects `A` and `B`, `A.timestamp ≠ B.timestamp`.
+  - Object timestamps are strictly increasing For objects `A` and `B`, if
+    `A.timestamp < B.timestamp`, then `A` was committed earlier than `B`.
+
+### Reply Order
+
+Replies to a client session always arrive in order — a client session may have only one request
+in-flight, and clients ignore (duplicate) replies to their prior requests.
+
+- Requests are executed in the order they arrive at the cluster's primary.
+- Replies to different clients may arrive out of order.
+
+#### Example
+
+Consider two clients `A` and `B`:
+
+  1. Client `A` sends request `A₁`.
+  2. Client `B` sends request `B₁`.
+
+Client `A` sent its request first, but requests `A₁` and `B₁` may execute in either order —
+whichever arrives first at the primary will execute first.
+
+In this diagram, the requests are delivered out of order — `B₁` then `A₁`:
+
+```mermaid
+sequenceDiagram
+    autonumber
+    participant Client A
+    participant Client B
+    participant (Network)
+    participant Cluster
+    Client A->>(Network): A₁ (request)
+    Client B->>(Network): B₁ (request)
+    Note over (Network): Request A₁ is delayed by the network.
+    (Network)->>Cluster: B₁ (request)
+    (Network)->>Cluster: A₁ (request)
+    Cluster->>(Network): B₁ (reply)
+    Cluster->>(Network): A₁ (reply)
+    (Network)->>Client B: B₁(reply)
+    (Network)->>Client A: A₁ (reply)
+```
+
+Suppose instead `A₁` arrives and executes before `B₁`.
+The replies may be delivered in the same order (`A₁` then `B₁`), or they may be reordered, as shown below:
+
+```mermaid
+sequenceDiagram
+    autonumber
+    participant Client A
+    participant Client B
+    participant (Network)
+    participant Cluster
+    Client A->>(Network): A₁ (request)
+    Client B->>(Network): B₁ (request)
+    (Network)->>Cluster: A₁ (request)
+    (Network)->>Cluster: B₁ (request)
+    Cluster->>(Network): A₁ (reply)
+    Cluster->>(Network): B₁ (reply)
+    Note over (Network): Reply A₁ is delayed by the network.
+    (Network)->>Client B: B₁(reply)
+    (Network)->>Client A: A₁ (reply)
+```
+
+### Retries
+
+A client will retry a request until either:
```

Comment:
"client" can be confusing on its own. Maybe we should define at the top when we say "client" we mean client library? Unless you actually mean user code? I'm not actually sure which you mean here.

Reply:
I'm still confused though. When you say "will" are you saying that TigerBeetle client libraries do this for you or that the application developer will have to put retries in place?

Reply:
TigerBeetle client libraries do this automatically.

How about:

> A client session will automatically retry a request until either: